### ISSUE TITLE
Investigate a broader credential type

### ISSUE THREAD

**Issue by alzimmermsft** [Azure/azure-sdk-for-java#40464]
While we can, we should investigate whether a broader credential interface could be created that spans the likes of KeyCredentail, NamedKeyCredential, etc to simplify the client builder story where services that support many authorization mechanisms would just have a single API for setting the authorization mechanism.

**Comment by JonathanGiles** [comment]
@alzimmermsft Is this still relevant?

**Comment by alzimmermsft** [comment]
No, I think this is well covered by policy changes, `HttpCredentialPolicy`, and we shouldn't unify credential types themselves.
